### DISCUSSION TITLE
这个项目适合用来做代码review吗？

### DISCUSSION THREAD

**Discussion by kerwin612** [Tencent/CodeAnalysis#538] [Q&A] [answered]
咨询下，这个项目是否支持`svn`的`pre-commit-hook`？
管理成员提交代码，做`checkstyle`、`findbugs`自动化检查；检查通过即可提交，不通过拒绝提交。

**Comment by bensonhome** [comment, marked as answer]
@kerwin612 是支持通过一行命令可以启动扫描的，可以配置hook来启动扫描，建议可以尝试：
1. 使用python3 codepuppy.py quickscan ... 命令扫描本地待提交的文件
2. 使用github action的方式，提交触发扫描：https://github.com/TCATools/TCA-action

**Comment by zhang9w0v5** [comment]
正如benson所说，在hook里调起分析进程就可以执行分析。和调起chechstyle或findbugs等是一样的。
在判断分析结果，依据结果进行处理，例如达到一定的指标就驳回提交。
